ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/EthConfig.java

```diff
@@ -65,23 +67,60 @@ public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
     long currentTime = System.currentTimeMillis() / 1000;
     ProtocolSpec current = protocolSchedule.getForNextBlockHeader(header, currentTime);
     Optional<ScheduledProtocolSpec> next = protocolSchedule.getNextProtocolSpec(currentTime);
+    Optional<ScheduledProtocolSpec> last = protocolSchedule.getLastProtocolSpec();
 
     ObjectNode result = mapperSupplier.get().createObjectNode();
     ObjectNode currentNode = result.putObject("current");
     generateConfig(currentNode, current);
-    result.put("currentHash", configHash(currentNode));
+    String currentHash = configHash(currentNode);
+    String nextHash = "";
+    String nextNextHash = "";
+    result.put("currentHash", currentHash);
     if (next.isPresent()) {
       ObjectNode nextNode = result.putObject("next");
       generateConfig(nextNode, next.get());
-      result.put("nextHash", configHash(nextNode));
+      nextHash = configHash(nextNode);
+      // get next from next
+      Optional<ScheduledProtocolSpec> nextNext =
+          protocolSchedule.getNextProtocolSpec(next.get().fork().milestone());
+      if (nextNext.isPresent()) {
+        ObjectNode nextNextNode =
+            mapperSupplier.get().createObjectNode(); // don't include in the result
+        generateConfig(nextNextNode, nextNext.get());
+        nextNextHash = configHash(nextNextNode);
+      }
+      result.put("currentForkId", getForkIdHashAsString(currentHash, nextHash));
+      result.put("nextHash", nextHash);
+      result.put("nextForkId", getForkIdHashAsString(nextHash, nextNextHash));
     } else {
+      // even if next was empty, still calculate currentForkId
+      result.put("currentForkId", getForkIdHashAsString(currentHash, nextHash));
       result.putNull("next");
       result.putNull("nextHash");
+      result.putNull("nextForkId");
+    }
+    if (last.isPresent()) {
+      ObjectNode lastNode = mapperSupplier.get().createObjectNode(); // don't include in the result
+      generateConfig(lastNode, last.get());
+      String lastHash = configHash(lastNode);
+      result.put("lastHash", lastHash);
+      result.put("lastForkId", getForkIdHashAsString(lastHash, ""));
+    } else {
+      // unexpected - last should always be present even if last == 0
```

Comment:
For a network with 1 fork, would we have current, empty next, last same as current?